app/Factories/AlbumFactory.php

```diff
@@ -84,6 +84,34 @@ public function findBaseAlbumOrFail(string $albumId, bool $withRelations = true)
 		}
 	}
 
+	/**
+	 * Returns an existing model instance of an album with the given ID or
+	 * fails with an exception.
+	 *
+	 * @param string $albumId       the ID of the requested album
+	 * @param bool   $withRelations indicates if the relations of an
+	 *                              album (i.e. photos and sub-albums,
+	 *                              if applicable) shall be loaded, too.
+	 *
+	 * @return Album the album for the ID
+	 *
+	 * @noinspection PhpIncompatibleReturnTypeInspection
+	 */
+	public function findAlbumOrFail(string $albumId, bool $withRelations = true): Album
```

Comment:
Why has this method been added?

Reply:
I need a method which only returns "real" albums - no tag albums. `findBaseAlbumOrFail` also returns tag albums.

Reply:
Well, but you don't need to fetch the album at all a second time. You can simply use the album object which is part of the request. See my comments on the request class below. So this whole method is not needed.